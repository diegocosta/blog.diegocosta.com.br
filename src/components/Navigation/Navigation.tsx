import React from 'react';
import styled from 'styled-components';

const Container = styled.nav`
  @media (min-width: 750px) {
    width: 400px;
  }
`;

const List = styled.ul`
  margin: 0;
  display: flex;
  padding: 0;
  justify-content: space-between;
`;

const Item = styled.li`
  list-style: none;
  font-size: 16px;
  font-weight: 700;

  @media (min-width: 760px) {
    font-size: 18px;
  }
`;

const Link = styled.a`
  text-decoration: none;
  color: #d73738;
  display: block;
  padding: 0;
  border-bottom: 1px solid transparent;

  :hover {
    border-bottom: 1px solid #d73738;
  }
`;

interface NavigationProps {
  list: [
    {
      link: string;
      label: string;
    }
  ];
}

export default function Navigation(props: NavigationProps): React.ReactElement {
  return (
    <Container>
      <List>
        {props.list &&
          props.list.map((item, index) => (
            <Item key={index}>
              <Link href={item.link} target="_blank" rel="noopener">
                {item.label}
              </Link>
            </Item>
          ))}
      </List>
    </Container>
  );
}
