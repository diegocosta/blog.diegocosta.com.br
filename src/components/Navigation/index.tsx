import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import Navigation from './Navigation';

export default (): React.ReactElement => {
  const {
    site: {
      siteMetadata: { contacts },
    },
  } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            contacts {
              label
              link
            }
          }
        }
      }
    `
  );

  return <Navigation list={contacts} />;
};
