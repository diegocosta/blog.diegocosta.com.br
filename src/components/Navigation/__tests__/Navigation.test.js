import React from 'react';
import { render } from '@testing-library/react';

import Navigation from '../Navigation';

describe('<Navigation />', () => {
  it('should render properly', () => {
    const contacts = [
      { link: 'mailto:johndoe@doe.com', label: 'email' },
      { link: 'https://google.com', label: 'site' },
    ];

    const { baseElement, queryByText } = render(<Navigation list={contacts} />);

    expect(queryByText('site')).toBeTruthy();
    expect(queryByText('email')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
