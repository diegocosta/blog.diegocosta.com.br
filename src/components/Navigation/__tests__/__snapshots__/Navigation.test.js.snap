// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render properly 1`] = `
.c1 {
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 {
  list-style: none;
  font-size: 16px;
  font-weight: 700;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #d73738;
  display: block;
  padding: 0;
  border-bottom: 1px solid transparent;
}

.c3:hover {
  border-bottom: 1px solid #d73738;
}

@media (min-width:750px) {
  .c0 {
    width: 400px;
  }
}

@media (min-width:760px) {
  .c2 {
    font-size: 18px;
  }
}

<body>
  <div>
    <nav
      class="c0"
    >
      <ul
        class="c1"
      >
        <li
          class="c2"
        >
          <a
            class="c3"
            href="mailto:johndoe@doe.com"
            rel="noopener"
            target="_blank"
          >
            email
          </a>
        </li>
        <li
          class="c2"
        >
          <a
            class="c3"
            href="https://google.com"
            rel="noopener"
            target="_blank"
          >
            site
          </a>
        </li>
      </ul>
    </nav>
  </div>
</body>
`;
